src/index.d.ts

```diff
@@ -0,0 +1 @@
+export * from './envalidWithDotenv'
```

Comment:
is this necessary? Wont `tsc` find the correct one based on `types` in package.json?

Reply:
I changed `types` in `package.json` to point to this file. I think it's more clear.

The other file meant to be imported (`indexNoDetenv.js`) also has an associated `.d.ts` file. Someone who is using a bundler that doesn't select the correct file can import it directly from `envalid/src/indexNoDetenv`.

Reply:
I still don't really understand why we need so many `.d.ts` files here (disclaimer– not a typescript user). Don't we need, at most, one per each entry point? The 4 here seem kind of excessive and I fear I might break it in the future without understanding why they're all necessary.

Reply:
I agree, addressed [here](https://github.com/af/envalid/pull/91#issuecomment-457998201).